csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -63,5 +63,68 @@ public int GetFieldIndex(string name, StringComparer comparer = default)
             return _fields.IndexOf(
                 _fields.Single(x => comparer.Compare(x.Name, name) == 0));
         }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || !this.GetType().Equals(obj.GetType()))
+            {
+                return false;
+            }
+            return Equals((Schema)obj);
+        }
+
+        public bool Equals(Schema other)
+        {
+            if (this == other)
+            {
+                return true;
+            }
+            var otherFields = other._fields;
+
+            if (this.HasMetadata != other.HasMetadata)
+            {
+                return false;
+            }
+            if (this._fields.Count != otherFields.Count)
+            {
+                return false;
+            }
+            
+            for (int ii = 0; ii < _fields.Count; ii++)
+            {
+                if (!_fields[ii].Equals(otherFields[ii]))
+                {
+                    return false;
+                }
+            }
+
+            if (this.HasMetadata && other.HasMetadata)
+            {
+                return this.Metadata.Keys.Count() == other.Metadata.Keys.Count() &&
+                       this.Metadata.Keys.All(k => other.Metadata.ContainsKey(k) && this.Metadata[k] == other.Metadata[k]) &&
+                       other.Metadata.Keys.All(k => this.Metadata.ContainsKey(k) && other.Metadata[k] == this.Metadata[k]);
+                //return this.Metadata.OrderBy(r => r.Key).SequenceEqual(other.Metadata.OrderBy(r => r.Key));
```

Comment:
In general, we shouldn't check in commented out code unless there is a super good reason for it.